src/util/tubemap.js

```diff
@@ -3918,6 +3918,26 @@ function trackDoubleClick() {
   createTubeMap();
 }
 
+// Takes a track and returns a string describing the nodes it passes through
+// In the format of >1>2<3>4, with the intergers being nodeIDs
+function getPathInfo(track) {
+  let result = "";
+  if (!track.sequence) {
+    return result;
+  }
+
+  for (const nodeID of track.sequence) {
+    // Node is approached backwards if "-" is present
+    if (Array.from(nodeID)[0] === "-") {
+      result = result.concat("<", nodeID.substring(1));
+    } else {
+      result = result.concat(">", nodeID);
```

Comment:
This looks like it is going to be an O(n^2) algorithm when it doesn't need to be. Building the new version of `result` is going to copy the original contents of `result` into a new string that also has the `<`/`>` character and the node ID. But that copy takes some unit of time for each character in `result`, and as you make `result` longer the number of characters to be copied increases at each step.

You probably want to build an array of all the pieces you want in the final string, by mutably appending to the array with `push()` (which doesn't take longer to do as the array gets longer). Then you can use `.join("")` on the array to turn it into a combined string all at once, in linear time.